### PR TITLE
task 4 completed

### PR DESCRIPTION
**Pull request by ojtaylor123**
Task 4 completed with testing

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -42,3 +42,37 @@ describe("GET /api/categories", () => {
...
+
+        expect(Array.isArray(reviews)).toBe(true);
+        expect(reviews.length).toBeGreaterThan(0);
+        expect(reviews[0] && typeof reviews[0] === "object").toBe(true);
+        expect(reviews).toBeSortedBy("created_at", {
+          descending: true,
+        });
+
+        expect(reviews.length).toBeGreaterThan(0);
+        expect(reviews[0] && typeof reviews[0] === "object").toBe(true);
```
Comment (Slyose) [inline __tests__/app.test.js]: These two lines are the same as 55 & 56

**Review by Slyose — APPROVED** [review]
Get rid of duplicate lines in test, then feel free to merge!

File: models/models.js [hunk 2]
```diff
@@ -9,4 +9,18 @@ exports.fetchCategories = ()=>{
...
+}
+
+exports.fetchReviews= () => {
+
+    return db.query(`
+
+    SELECT owner, title,reviews.review_id, category, review_img_url, reviews.created_at, reviews.votes, designer, COUNT(comments.review_id)
+    FROM reviews
+    LEFT JOIN comments ON reviews.review_id = comments.review_id
+    GROUP BY owner, title, category, review_img_url, reviews.created_at, reviews.votes, designer,reviews.review_id
```
Comment (xViolaine) [inline models/models.js:21]: You only need to be grouping by reviews.review_id here

File: __tests__/app.test.js (hunk 1 above)
Comment (ojtaylor123) [reply __tests__/app.test.js]: thanks

File: models/models.js [hunk 3]
```diff
@@ -9,4 +9,18 @@ exports.fetchCategories = ()=>{
         return result.rows
     })
 
+}
+
+exports.fetchReviews= () => {
+
+    return db.query(`
+
+    SELECT owner, title,reviews.review_id, category, review_img_url, reviews.created_at, reviews.votes, designer, COUNT(comments.review_id)
```
Comment (xViolaine) [inline models/models.js:18]: Count is a good choice here, maybe have a look at COUNT .. AS - after all, we do want this to be called comment_count

**Review by xViolaine — CHANGES_REQUESTED** [review]

File: __tests__/app.test.js [hunk 4]
```diff
@@ -42,3 +42,34 @@ describe("GET /api/categories", () => {
...
+        reviews.forEach((review) => {
+          expect(review).toMatchObject({
+            owner: expect.any(String),
+            title: expect.any(String),
+            review_id: expect.any(Number),
+            category: expect.any(String),
+            review_img_url: expect.any(String),
+            created_at: expect.any(String),
+            votes: expect.any(Number),
+            count: expect.any(String),
```
Comment (Slyose) [inline __tests__/app.test.js:70]: This property you're testing for on line 70 should be comment_count not just count.

**Pull request merged**